packages/registry/README.md

```diff
@@ -0,0 +1,190 @@
+# Add-on and configuration registry
+
+This package provides support for building an add-on registry and infrastructure for JS/TS-based apps.
+
+## Extensibility and puggability
+
+As developers, when you build an app, regardless of the framework and technologies used, it's a one-off app. Meaning, you have to build something that has very specific requirements, behavior and look and feel.
+
+Other few times, you need to build something generic, that could be pluggable and extensible. In the JS/TS ecosystem this is often quite complex and the existing frameworks do not provide the meanings to "build up" in these regards.
+
+## Add-on registry
+
+An add-on registry is a facility that allows an app built on an existing framework to build on extensibility and pluggability.
+
+The add-on registry is a store where you can register a number of add-ons that your app is going to consume.
+
+Add-on packages are just CommonJS packages. The only requirement is that
+they point the `main` key of their `package.json` to a module that exports, as a default function that acts as a configuration loader.
+
+An add-on can be published in an npm registry, just as any other package. However, add-ons are meant to not be transpiled. They should be released as "source" packages.
+
+## Register an add-on
+
+You should declare in your project that you are using an add-on.
+This is done in the åpp's `package.json`, `addons` key:
+
+```json
+{
+  "name": "my-app-project",
+  "addons": [
+    "acme-volto-foo-addon",
+    "@plone/some-addon",
+    "collective-another-volto-addon"
+  ]
+}
+```
+
+By including the add-on name in the `addons` key, the add-on's main default export function is executed, being passed the configuration registry. In that function, the add-on can customize the registry. The function needs to return the `config` (configuration registry) object, so that it's passed further along to the other add-ons.
```

Comment:
```suggestion
The `addons` key ensures the add-on's main default export function is executed, being passed the configuration registry.
In that function, the add-on can customize the registry.
The function needs to return the `config` (configuration registry) object, so that it's passed further along to the other add-ons.
```